libs/payments/stripe/src/lib/stripe.util.spec.ts

```diff
@@ -206,4 +211,48 @@ describe('util', () => {
       expect(result).toEqual([]);
     });
   });
+
+  describe('stripeInvoiceToFirstInvoicePreviewDTO', () => {
+    it('formats invoice', () => {
+      const mockUpcomingInvoice = StripeResponseFactory(
+        StripeUpcomingInvoiceFactory({
+          total_discount_amounts: [
+            {
+              amount: 500,
+              discount: StripeDiscountFactory(),
+            },
+          ],
+          total_tax_amounts: [
+            {
+              amount: faker.number.int(1000),
+              inclusive: false,
+              tax_rate: StripeTaxRateFactory(),
+              taxability_reason: null,
+              taxable_amount: null,
+            },
+          ],
+        })
+      );
+
+      const result = stripeInvoiceToFirstInvoicePreviewDTO(mockUpcomingInvoice);
+      expect(result).toEqual({
+        currency: mockUpcomingInvoice.currency,
+        listAmount: mockUpcomingInvoice.amount_due,
+        totalAmount: mockUpcomingInvoice.total,
+        taxAmounts: [
+          {
+            title:
+              mockUpcomingInvoice.total_tax_amounts[0].tax_rate.display_name,
+            inclusive: mockUpcomingInvoice.total_tax_amounts[0].inclusive,
+            amount: mockUpcomingInvoice.total_tax_amounts[0].amount,
+          },
+        ],
+        discountAmount:
+          mockUpcomingInvoice.total_discount_amounts &&
+          mockUpcomingInvoice.total_discount_amounts.length > 0
+            ? mockUpcomingInvoice.total_discount_amounts[0].amount
+            : null,
```

Comment:
Not sure if it matters, but this logic looks subtly different than the code above in libs/payments/cart/src/lib/cart.service.spec.ts:

```js
      discountAmount:
        mockUpcomingInvoice.total_discount_amounts &&
        mockUpcomingInvoice.total_discount_amounts[0].amount,
```

So here, we're adding an additional check for `.length > 0` and then using that in a ternary to default to `null` (verus, I think, defaulting to `undefined` in the version in libs/payments/cart/src/lib/cart.service.spec.ts). They're both just .spec.ts files so might be moot; but I'm a big believer in "see something, file annoying GitHub comments". 🤷